paddle/fluid/framework/ir/conv_bn_fuse_pass.cc

```diff
@@ -0,0 +1,331 @@
+    GET_IR_NODE_FROM_SUBGRAPH(bn_saved_variance, bn_saved_variance,
+                              conv_bn_pattern);
+
+    // Find bias from BN
+    auto* bn_bias_var = scope->FindVar(bn_bias->Name());
+    auto* bn_bias_tensor = bn_bias_var->GetMutable<LoDTensor>();
+
+    // Get y (bias) variable
+    auto* bias_y_tensor =
+        scope->FindVar(bias_y->Name())->GetMutable<LoDTensor>();
+
+    PADDLE_ENFORCE_EQ(bias_y_tensor->dims(), bn_bias_tensor->dims());
+
+    // Compute new weights and bias
+    auto* scale_tensor =
+        scope->FindVar(bn_scale->Name())->GetMutable<LoDTensor>();
+    auto* variance_tensor =
+        scope->FindVar(bn_variance->Name())->GetMutable<LoDTensor>();
+    auto* mean_tensor =
+        scope->FindVar(bn_mean->Name())->GetMutable<LoDTensor>();
+
+    auto std_tensor = LoDTensor();
+    std_tensor.Resize(bn_bias_tensor->dims());
+    std_tensor =
+        tensor_apply(variance_tensor, [](float x) { return x + 1e-5f; });
+
+    tensor_apply_inplace(&std_tensor, std::sqrt);
+    auto tmp_tensor =
+        tensor_apply_eltwise(*scale_tensor, std_tensor, std::divides<float>());
+    {
+      auto tensor_minus = tensor_apply_eltwise(*bias_y_tensor, *mean_tensor,
+                                               std::minus<float>());
+      auto tensor_mul = tensor_apply_eltwise(tensor_minus, tmp_tensor,
+                                             std::multiplies<float>());
+      *bias_y_tensor =
+          tensor_apply_eltwise(tensor_mul, *bn_bias_tensor, std::plus<float>());
+    }
+
+    // Re-compute weight of conv2d from BN
+    auto* current_param =
+        scope->FindVar(conv_weight->Name())->GetMutable<LoDTensor>();
+    // remember the weights tensor shape {A, B, C, ...}
+    auto current_param_shape = current_param->dims();
+    // flatten the weights tensor shape {A, B * C * ...}
+    if (current_param->dims().size() > 1) {
+      int size2 = current_param->dims()[1];
+      for (int i = 2; i < current_param->dims().size(); i++) {
+        size2 *= current_param->dims()[i];
+      }
+      current_param->Resize(make_ddim({current_param->dims()[0], size2}));
+    }
+    *current_param = tensor_apply_eltwise_broadcast(*current_param, tmp_tensor,
+                                                    std::multiplies<float>());
+    // reshape weights to the original dims {A, B, C, ...}
+    current_param->Resize(current_param_shape);
+
+    // Modify bias of elementwise add node.
+    bias->Op()->SetAttr("axis", 1);
+    bias->Op()->SetOutput("Out", std::vector<std::string>({bn_out->Name()}));
+
+    GraphSafeRemoveNodes(
+        graph.get(),
+        {bn_scale, bn_bias, bn_mean, bn_variance, batch_norm, bn_mean_out,
+         bn_variance_out, bn_saved_mean, bn_saved_variance, bias_out});
+
+    PADDLE_ENFORCE(subgraph.count(conv_input));
+    IR_NODE_LINK_TO(bias, bn_out);
+
+    found_conv_bn_count++;
+  };
+
+  gpd(graph.get(), handler);
+
+  AddStatis(found_conv_bn_count);
+  return graph;
+}
```

Comment:
`ConvEltwiseAddBNFusePass` and `ConvBNFusePass` have many similar codes except `Get y (bias) variable`. Could you abstract a common function to simplify these two functions?

Reply:
Ok.